src/platform/forms-system/src/js/fields/TitleField.jsx

```diff
@@ -8,7 +8,7 @@ export default function TitleField({ id, title }) {
     'schemaform-block-subtitle': !isRoot,
   });
 
-  return (
+  return typeof title === 'string' && title.trim() === '' ? null : (
```

Comment:
Can you pull this out into a variable so we can give it a name that helps explain what it's doing? `titleIsBlank` or something like that.